src/TLuaInterpreter.cpp

```diff
@@ -1189,19 +1189,34 @@ int TLuaInterpreter::getMapEvents(lua_State * L){
     {
         if (host.mpMap->mpMapper){
             if (host.mpMap->mpMapper->mp2dMap){
+                // create the result table
                 lua_newtable(L);
                 QMapIterator<QString, QStringList> it(host.mpMap->mpMapper->mp2dMap->mUserActions);
                 while (it.hasNext()){
                     it.next();
-                    lua_newtable(L);
                     QStringList eventInfo = it.value();
-                    lua_pushstring( L, eventInfo[0].toLatin1().data() );
-                    lua_pushstring( L, eventInfo[1].toLatin1().data() );
-                    lua_pushstring( L, eventInfo[2].toLatin1().data() );
-                    lua_settable(L, -3);
-                    lua_pushstring(L, it.key().toLatin1().data());
-                    lua_insert(L,-2);
-                    lua_settable(L, -3);
+
+                    lua_createtable( L, 0, 4 );
+
+                    lua_pushstring( L, eventInfo[0].toUtf8().constData() );
```

Comment:
Since we are only reading the `QStringList` it is better to use the (faster/safer) read-only `QStringList::at(int)` compared to the read-write `QStringList::operator[int]`... IMNSHO 🤓